models/bridge/zksync/bridge_zksync_native_flows.sql

```diff
@@ -0,0 +1,174 @@
+{{ config(
+    schema = 'bridge_zksync_native',
+    alias = 'flows',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["zksync"]\',
+                                "sector",
+                                "bridge",
+                                \'["lgingerich"]\') }}'
+    )
+}}
+
+WITH bridge_events as (
+    SELECT
+         block_time
+        ,CAST(DATE_TRUNC('day', block_time) as date) as block_date
+        ,CAST(DATE_TRUNC('month', block_time) as date) as block_month
+        ,block_number
+        ,tx_hash
+        ,evt_index
+        ,sender_address
+        ,receiver_address
+        ,bridged_token_address
+        ,bridged_token_amount_raw
+        ,source_chain_id
+        ,destination_chain_id
+        ,source_chain_name
+        ,destination_chain_name
+        ,tx_from
+        ,tx_to
+    FROM (
+        -- Deposit ETH from Ethereum to zkSync Era
+        SELECT
+             npr.evt_block_time as block_time
+            ,npr.evt_block_number as block_number
+            ,npr.evt_tx_hash as tx_hash
+            ,npr.evt_index
+            ,et."from" as sender_address
+            ,COALESCE(d.to, zt.to) as receiver_address -- d.to is null if there is no matching ERC20 deposit tx hash. This is used to handle the logic of an 'if statement'.
+            ,0x0000000000000000000000000000000000000000 as bridged_token_address
+            ,CAST(JSON_EXTRACT_SCALAR(npr.transaction, '$.reserved[0]') as UINT256) as bridged_token_amount_raw
+            ,UINT256 '1' as source_chain_id
+            ,UINT256 '324' as destination_chain_id
+            ,'ethereum' as source_chain_name
+            ,'zksync' as destination_chain_name
+            ,et."from" as tx_from
+            ,et.to as tx_to
+        FROM {{ source('ethereum', 'transactions') }} et
+        INNER JOIN {{ source('zksync_v2_ethereum', 'DiamondProxy_evt_NewPriorityRequest') }} npr ON et.hash = npr.evt_tx_hash
```

Comment:
```suggestion
        INNER JOIN {{ source('zksync_v2_ethereum', 'DiamondProxy_evt_NewPriorityRequest') }} npr
        ON et.hash = npr.evt_tx_hash
        {% if is_incremental() %}
        AND {{ incremental_predicate('npr.evt_block_time') }}
        {% endif %}
```

Reply:
Sorry, I missed your suggestion above with this same suggestion. Yes I will add it

Reply:
this is what i mean. the core idea is:
- all tables filter to same date ranges on read, this speeds up hourly job refreshes in prod
- "base" tables in the from will be in a where clause
- joined tables typically within the join conditions
  - technically if an inner join, then could filter in where clause too, but it's easier to read in joins
  - left/right joins it's more efficient to have in join conditions for sure

Reply:
you can apply this throughout the entire spell on each union

Reply:
Ok I think I'm understanding you now. Let me know if my new commit covers this properly